### PR TITLE
Add reference store integration for StructureResolver

### PR DESCRIPTION
**Pull request by alexander-schranz**

### CODE REVIEW

**Comment by alexander-schranz** [comment]
When implementing #79 I thought it would be better to move the reference store to the serializer services. There sadly does not exist a `PageSerializer`, there the general StructureResolver is used, which I totally forget in all cases need a referenceStore so the page itself can be invalidated by the tag. Sadly the StructureResolver would need to know multiple reference store. How do we handle this? I could go on the StructureType or what is called but would the reference store for the pages was not updated and still use the old `content` name so I would to implement a mapping:

 - article -> article
 - snippet -> snippet
 - content -> page

**Review by wachterjohannes — APPROVED** [review]

File: Content/StructureResolver.php [hunk 1]
```diff
@@ -58,6 +67,9 @@ public function resolve(
     ): array {
         $data = $this->getStructureData($structure);
 
+        /** @var BasePageDocument $document */
+        $document = $structure->getDocument();
+
         if ($includeExtension) {
```
Comment (niklasnatter) [inline Content/StructureResolver.php]:
```suggestion
        if ($includeExtension) {
```

File: Content/StructureResolver.php [hunk 2]
```diff
@@ -289,4 +303,21 @@ private function resolveProperty(
             $attributes
         );
     }
+
+    private function addToReferenceStore(string $uuid, string $alias): void
+    {
+        if ('page' === $alias) {
+            $alias = 'content'; // the reference store for page sadly is called still content
```
Comment (niklasnatter) [inline Content/StructureResolver.php]:
```suggestion
            // unfortunately the reference store for pages was not adjusted and still uses the content as alias
            $alias = 'content';
```

File: Tests/Unit/Content/StructureResolverTest.php [hunk 3]
```diff
@@ -433,6 +477,14 @@ public function testResolvePropertiesIncludeExtension(): void
         $this->contentResolver->resolve('test-123', $titleProperty->reveal(), 'en', ['webspaceKey' => 'sulu_io'])
             ->willReturn($contentView1->reveal());
 
+        $this->referenceStorePool->getStore('content')
```
Comment (niklasnatter) [inline Tests/Unit/Content/StructureResolverTest.php]: can we use another type (eg. `snippet(s)`) in one of the test cases? 🤔

Comment (alexander-schranz) [reply Tests/Unit/Content/StructureResolverTest.php]: added!

File: Content/StructureResolver.php [hunk 4]
```diff
@@ -289,4 +300,22 @@ private function resolveProperty(
             $attributes
         );
     }
+
+    private function addToReferenceStore(string $uuid, string $alias): void
+    {
+        if ('page' === $alias) {
+            // unfortunately the reference store for pages was not adjusted and still uses the content as alias
```
Comment (niklasnatter) [inline Content/StructureResolver.php]:
```suggestion
            // unfortunately the reference store for pages was not adjusted and still uses content as alias
```

**Review by niklasnatter — APPROVED** [review]

**Pull request merged**